File: lib/audio_engine_player.dart
import 'dart:async';
import 'audio_engine_player_platform_interface.dart';

enum PlayerEventType {
  playbackProgress,
  playingStatus,
  playCompleted,
  playingIndex
}

class AudioEnginePlayer {
  Future<String?> get platformVersion async {
    return AudioEnginePlayerPlatform.instance.getPlatformVersion();
  }

  Future<int> get duration async {
    return AudioEnginePlayerPlatform.instance.duration();
  }

  Future<double> get volume async {
    return AudioEnginePlayerPlatform.instance.volume();
  }

  Future<bool> get isMute async {
    return AudioEnginePlayerPlatform.instance.isMute();
  }

  Future<int> get currentPlayIndex async {
    return AudioEnginePlayerPlatform.instance.currentPlayIndex();
  }

  Future<bool> get isPlaying async {
    return AudioEnginePlayerPlatform.instance.isPlaying();
  }

  Future<void> ensureEngineRunning() async {
    return AudioEnginePlayerPlatform.instance.ensureEngineRunning();
  }

  Future<void> playWith(
      String filePath, String title, String artist, String album) {
    return AudioEnginePlayerPlatform.instance
        .playWith(filePath, title, artist, album);
  }

  Future<void> seekTo(int milliseconds) async {
    return AudioEnginePlayerPlatform.instance.seekTo(milliseconds);
  }

  Future<void> seekToIndex(int index) async {
    return AudioEnginePlayerPlatform.instance.seekToIndex(index);
  }

  Future<void> playOrPause() async {
    return AudioEnginePlayerPlatform.instance.playOrPause();
  }

  Future<void> play() async {
    return AudioEnginePlayerPlatform.instance.play();
  }

  Future<void> pause() async {
    return AudioEnginePlayerPlatform.instance.pause();
  }

  Future<void> stop() async {
    return AudioEnginePlayerPlatform.instance.stop();
  }

  Future<void> setPlaylist(List<Map<String, String>> tracks, bool autoPlay) {
    return AudioEnginePlayerPlatform.instance.setPlaylist(tracks, autoPlay);
  }

  Future<void> appendToPlaylist(
      String source, String title, String artist, String album, bool autoPlay) {
    return AudioEnginePlayerPlatform.instance
        .appendToPlaylist(source, title, artist, album, autoPlay);
  }

  Future<void> removeFromPlaylist(int index) async {
    return AudioEnginePlayerPlatform.instance.removeFromPlaylist(index);
  }

  Future<void> moveOnPlaylist(int oldIndex, int newIndex) async {
    return AudioEnginePlayerPlatform.instance
        .moveOnPlaylist(oldIndex, newIndex);
  }

  Future<void> playNext() async {
    return AudioEnginePlayerPlatform.instance.playNext();
  }

  Future<void> playPrevious() async {
    return AudioEnginePlayerPlatform.instance.playPrevious();
  }

  Future<void> setSpeed(double value) async {
    return AudioEnginePlayerPlatform.instance.setSpeed(value);
  }

  Future<void> setVolume(double value) async {
    return AudioEnginePlayerPlatform.instance.setVolume(value);
  }

  Future<void> setVolumeBoost(double value) async {
    return AudioEnginePlayerPlatform.instance.setVolumeBoost(value);
  }

  Future<void> setMute(bool value) async {
    return AudioEnginePlayerPlatform.instance.setMute(value);
  }

  Future<void> setEnableFadeEffect(bool value) async {
    return AudioEnginePlayerPlatform.instance.setEnableFadeEffect(value);
  }

  Future<void> setLoopMode(int mode) async {
    return AudioEnginePlayerPlatform.instance.setLoopMode(mode);
  }

  Future<void> setBandGain(int bandIndex, double gain) async {
    return AudioEnginePlayerPlatform.instance.setBandGain(bandIndex, gain);
  }

  Future<void> setReverb(int id, double wetDryMix) async {
    return AudioEnginePlayerPlatform.instance.setReverb(id, wetDryMix);
  }

  Future<void> resetAll() async {
    return AudioEnginePlayerPlatform.instance.resetAll();
  }

  Future<void> clearCaches() async {
    return AudioEnginePlayerPlatform.instance.clearCaches();
  }

  Stream<(String, Map<String, dynamic>)> get onPlayerEvents {
    return AudioEnginePlayerPlatform.instance.onEventStream
        .where((event) => event['event'] != null)
        .map((event) {
      try {
        final eventType = event['event'] as String;
        final eventData = Map<String, dynamic>.from(event)..remove('event');
        return (eventType, eventData);
      } catch (e) {
        return ('error', {'desc': e.toString()});
      }
    });
  }
}

File: android/src/main/kotlin/com/ppsw/tech/audio_engine_player/AudioEnginePlayer.kt
package com.ppsw.tech.audio_engine_player

enum class LoopMode {
  ALL, SINGLE, SHUFFLE
}

class AudioEnginePlayer {
  var totalDuration: Long = 0
  var volume: Float = 1.0f

  fun play(filePath: String) {
    // 实现播放逻辑
  }

  fun seekTo(milliseconds: Int) {
    // 实现跳转逻辑
  }

  fun playOrPause() {
    // 实现播放或暂停逻辑
  }

  fun stop() {
    // 实现停止逻辑
  }

  fun setPlaylist(urls: List<String>, autoPlay: Boolean) {
    // 实现设置播放列表逻辑
  }

  fun appendToPlaylist(url: String, autoPlay: Boolean) {
    // 实现追加到播放列表逻辑
  }

  fun playNext() {
    // 实现播放下一首逻辑
  }

  fun playPrevious() {
    // 实现播放上一首逻辑
  }

  fun setVolume(volume: Float) {
    // 实现设置音量逻辑
  }

  fun setLoopMode(loopMode: LoopMode) {
    // 实现设置循环模式逻辑
  }

  fun setBandGain(bandIndex: Int, gain: Float) {
    // 实现设置频段增益逻辑
  }

  fun setReverb(id: Int, wetDryMix: Float) {
    // 实现设置混响逻辑
  }

  fun resetAll() {
    // 实现重置所有设置逻辑
  }

  fun clearCaches() {
    // 实现清除缓存逻辑
  }
}